armi/reactor/assemblies.py

```diff
@@ -1317,6 +1317,37 @@ def getSymmetryFactor(self):
         """Return the symmetry factor of this assembly."""
         return self[0].getSymmetryFactor()
 
+    def rotate(self, deg):
+        # note this can only rotate in 60 degree increments.
+        # as this is only called by growToFullCore which is strictly hex, the 60 degree restriction is acceptable
+        rotNum = round((deg % (2 * math.pi)) / math.radians(60))
+        for b in self.getBlocks():
+            b.rotatePins(rotNum)
+            params = b.p.paramDefs.atLocation(ParamLocation.CORNERS).names
+            params += b.p.paramDefs.atLocation(ParamLocation.EDGES).names
+            for param in params:
+                if isinstance(b.p[param], list):
+                    if len(b.p[param]) == 6:
+                        b.p[param] = b.p[param][-rotNum:] + b.p[param][:-rotNum]
+                    elif b.p[param] == []:
+                        # List hasn't been defined yet, no warning needed.
+                        pass
+                    else:
+                        runLog.warning(
+                            "No rotation method defined for spatial parameters that aren't defined once per hex edge/corner. No rotation performed on {}".format(
+                                param
+                            )
+                        )
+                else:
+                    # this is a scalar and there shouldn't be any rotation.
+                    pass
+
+            dispx = b.p.get("displacementX")
+            dispy = b.p.get("displacementY")
```

Comment:
should just use the normal `b.p.displacementX` syntax so IDEs can track the variables.

Reply:
I have updated the other calls to use b.p.displacementX. In this particular case, I left it as is because I specifically used the .get() function in case the parameter does not exist. The get function will return None which will be caught by the next if statement. I think if I were to access by the .displacementX and it didn't exist, I would get an error?